### PR TITLE
Reorganize overview doc

### PR DESCRIPTION
**Pull request by ktock**
Fixes: #91

This commit includes:
- Moved registry-related config from README to the overview doc
- Added a section about filesystem (including hidden state directory)

### CODE REVIEW

File: docs/overview.md [hunk 1]
```diff
@@ -1,59 +1,151 @@
...
+  [proxy_plugins.stargz]
+    type = "snapshot"
+    address = "/run/containerd-stargz-grpc/containerd-stargz-grpc.sock"
 
-Because file contents are fetched via NW on each access, read performance would be one of the major concerns.
-To mitigate it, Stargz Snapshotter provides additional workload-oriented optimization.
-When you convert an image to stargz format using `ctr-remote`, you can specify some options which describe your workload  (i.e. entrypoint commands, environment variables, etc.).
-For example, we can optimize `ubuntu:18.04` image for execution of `ls` command on `bash` as following,
+# Use stargz snapshotter through CRI
+[plugins."io.containerd.grpc.v1.cri".containerd]
```
Comment (AkihiroSuda) [inline docs/overview.md:45]: requires `version = 2` header

File: docs/overview.md [hunk 2]
```diff
@@ -1,59 +1,151 @@
...
+  snapshotter = "stargz"
+```
+
+This repo contains [a Dockerfile as a KinD node image](/Dockerfile) which includes the above configuration.
+
+## The filesystem
```
Comment (AkihiroSuda) [inline docs/overview.md:51]: `## State directory`

File: docs/overview.md [hunk 3]
```diff
@@ -1,59 +1,151 @@
...
+root@1d43741b8d29:/go# cat /.stargz-snapshotter/*
+{"digest":"sha256:2b1fc65cafe05b65acc9e9f186df4dd81ae74c58ef73d89ecfc15e7286b3e960","size":131339690,"fetchedSize":7939690,"fetchedPercent":6.045156646859757}
+{"digest":"sha256:42d56485c1f672e394a02855048774621731c8fd44a54dc816a421a3a52b8482","size":10047608,"fetchedSize":2047608,"fetchedPercent":20.379059374131632}
+{"digest":"sha256:6a5826d877de5c93fb4a9e1d0369cfdef6d43df2610562501ebf42e4bcb2ef73","size":54352828,"fetchedSize":2302828,"fetchedPercent":4.236813584014432}
+{"digest":"sha256:a4d35801573274df19d9c2ae2aed80eba96d5aa69a38c464e1f01f9abf81e34e","size":70359295,"fetchedSize":2259295,"fetchedPercent":3.211082487395588}
+{"digest":"sha256:ab13100112faac6e04d2da2281db3df942efc8cef2532ab2cac688c6232944d8","size":7890588,"fetchedSize":2140588,"fetchedPercent":27.12837116828302}
+{"digest":"sha256:e8cc31024eb09fe216ad906392aec139038330c6d29dfd3fe5c81c4b2dd21430","size":52934435,"fetchedSize":2634435,"fetchedPercent":4.976788738748227}
+{"digest":"sha256:f077511be7d385c17ba88980379c5cd0aab7068844dffa7a1cefbf68cc3daea3","size":580,"fetchedSize":580,"fetchedPercent":100}
+```
+
```
Comment (AkihiroSuda) [inline docs/overview.md:90]: Make sure to note that the state directory layout is still subject to change

File: docs/overview.md [hunk 4]
```diff
@@ -1,59 +1,151 @@
...
+You can configure stargz snapshotter for accessing registries with custom configurations.
+The config file must be formatted with TOML and can be passed to stargz snapshotter with `--config` option.
+
+### Authentication
+
+Stargz snapshotter doesn't share private registries creds with containerd.
+Instead, this supports authentication in the following methods,
 
+- Using `DOCKER_CONFIG` or `~/.docker/config.json`
```
Comment (AkihiroSuda) [inline docs/overview.md:101]: `$DOCKER_CONFIG`

File: docs/overview.md [hunk 5]
```diff
@@ -1,59 +1,151 @@
...
-             ubuntu:18.04 http://registry2:5000/ubuntu:18.04
+
+Following configuration enables stargz snapshotter to access to private registries using kubernetes secrets (type = `kubernetes.io/dockerconfigjson`) in the cluster using kubeconfig files.
+You can specify the path of kubeconfig file using `kubeconfig_path` option.
+It's no problem that the specified file doesn't exist when this snapshotter starts.
+In this case, snapsohtter polls the file until actually provided.
+This is useful for some environments (e.g. single node cluster with containerized apiserver) where stargz snapshotter needs to start before everything, including booting containerd/kubelet/apiserver and configuring users/roles.
+If no `kubeconfig_path` is specified, snapshotter searches kubeconfig files from `KUBECONFIG` or `~/.kube/config`.
+
+```toml
```
Comment (AkihiroSuda) [inline docs/overview.md:121]: The file path?

Comment (ktock) [reply docs/overview.md:121]: Could you elaborate this comment? Should we describe the file path that this config is written to?

Comment (AkihiroSuda) [reply docs/overview.md:121]: yes for clarity

Comment (ktock) [reply docs/overview.md:125]: Thanks for the feedback.
`containerd-stargz-grpc` currently doesn't have the default config file path. 
So I added the description `The config file can be passed to stargz snapshotter using containerd-stargz-grpc's --config option.`.

**Pull request merged**